resources/application-connector/charts/connector-service/templates/helm-certs-removal-backup.yaml

```diff
@@ -0,0 +1,98 @@
+{{- range $key, $value := .Values.certs -}}
+{{- $_ := set $ "key" $key }}
+{{- $_ := set $ "value" $value }}
+---
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: {{ $value.name }}-certs-backup
+  namespace: {{ $value.namespace }}
+  annotations:
+    helm.sh/hook: pre-upgrade
+    helm.sh/hook-delete-policy: before-hook-creation
+    helm.sh/hook-weight: "1"
+  labels:
+    release: {{ $.Release.Name }}
+    helm.sh/chart: {{ $.Chart.Name }}-{{ $.Chart.Version | replace "+" "_" }}
+    app.kubernetes.io/name: {{ template "name" $ }}
+    app.kubernetes.io/managed-by: {{ $.Release.Service }}
+    app.kubernetes.io/instance: {{ $.Release.Name }}
+spec:
+  template:
+    metadata:
+      name: {{ $value.name }}-helm-certs-removal-backup
+      namespace: {{ $value.namespace }}
+      annotations:
+        sidecar.istio.io/inject: "false"
+      labels:
+        release: {{ $.Release.Name }}
+        helm.sh/chart: {{ $.Chart.Name }}-{{ $.Chart.Version | replace "+" "_" }}
+        app.kubernetes.io/name: {{ template "name" $ }}
+        app.kubernetes.io/managed-by: {{ $.Release.Service }}
+        app.kubernetes.io/instance: {{ $.Release.Name }}
+    spec:
+      serviceAccountName: {{ $value.name }}-helm-certs-removal-acc
+      restartPolicy: Never
+      {{- if $.Values.global.priorityClassName }}
+      priorityClassName: {{ $.Values.global.priorityClassName }}
+      {{- end }}
+      containers:
+        - name: {{ $value.name }}-helm-certs-removal-backup
+          image: "{{ $.Values.jobs.image.repository }}:{{ $.Values.jobs.image.tag }}"
+          securityContext:
+            runAsUser: {{ $.Values.global.podSecurityPolicy.runAsUser }}
+            privileged: {{ $.Values.global.podSecurityPolicy.privileged }}
+            allowPrivilegeEscalation: {{ $.Values.global.podSecurityPolicy.allowPrivilegeEscalation }}
+          command:
+            - /bin/bash
+            - -c
+            - |-
+{{ tpl ($.Files.Get "files/certs-backup.sh" | printf "%s" | indent 14) $ }}
+---
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: {{ $value.name }}-helm-certs-removal-restore
+  namespace: {{ $value.namespace }}
+  annotations:
+    helm.sh/hook: post-upgrade
+    helm.sh/hook-delete-policy: before-hook-creation
+    helm.sh/hook-weight: "1"
```

Comment:
I'm not sure if the flag value is correct. Certs Setup Job has no `helm.sh/hook-weight `annotation set so weight value is default (0). Correct me if I'm wrong, but the Certs Setup Job will be executed _before_ restore job (https://helm.sh/docs/topics/charts_hooks/). Preferred order is to restore first.